src/menu.ts

```diff
@@ -787,6 +787,13 @@ export class Menu<C extends Context = Context> extends MenuRange<C>
             ctx,
             async (btn, i, j): Promise<InlineKeyboardButton> => {
                 const text = await uniform(ctx, btn.text);
+                
+                if ("url" in btn) {
+                    let {url, ...rest} = btn;
+                    url = await uniform(ctx, btn.url);
+                    return { ...rest, url, text };
+                }
+                
                 if ("middleware" in btn) {
```

Comment:
Let's either do
```suggestion
                } else if ("middleware" in btn) {
```
or remove the trailing `else if` but not mix two different styles :D